app/src/test/java/waiter/CommunicatorTest.java

```diff
@@ -0,0 +1,21 @@
+package waiter;
+
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class CommunicatorTest {
+
+    @Test
+    void establishesCommunication() throws IOException {
+        String userInput = "Doesn't matter whats here, simulating any user input message";
+        ListenerMock listenerMock = new ListenerMock(userInput);
+        MessengerMock messengerMock = new MessengerMock();
+        Communicator communicator = new Communicator();
+        communicator.communicate(listenerMock, messengerMock);
```

Comment:
```suggestion
        Communicator communicator = new Communicator(listenerMock, messengerMock);
        communicator.communicate();
```

Did you consider moving the `listener` and `messenger` into the `Communicator` constructor? Do we need to be able to specify different listeners/messengers when calling `communicate`?

Reply:
That is a really good point, we can move them into the constructor, not sure why I didn't in the first place, thank you 😄

Reply:
Resolved in [this commit](https://github.com/jonmanahan/waiter/pull/2/commits/e53fdbc1eec22480eae57e79838d0f1186c16f90), if it resolves the issues please hit resolve conversation if possible, thank you all a ton!